overlord/state/taskrunner.go

```diff
@@ -469,6 +491,15 @@ ConsiderTasks:
 		r.state.EnsureBefore(nextTaskTime.Sub(ensureTime))
 	}
 
+	// run hooks registered for task exhaustion
+	if len(running) == 0 {
+		for _, hooks := range r.hooks {
+			for _, h := range hooks {
```

Comment:
shouldn't this be _, h := range r.hooks[TaskExhaustionHook]

Reply:
Super nice catch!